templates/types/streaming/nextjs/app/components/ui/upload-pdf-preview.tsx

```diff
@@ -0,0 +1,86 @@
+import { FileText, XCircleIcon } from "lucide-react";
+import { Button } from "./button";
+import { PdfFile } from "./chat";
+import {
+  Drawer,
+  DrawerClose,
+  DrawerContent,
+  DrawerDescription,
+  DrawerHeader,
+  DrawerTitle,
+  DrawerTrigger,
+} from "./drawer";
+import { cn } from "./lib/utils";
+
+export interface UploadPdfPreviewProps {
+  pdf: PdfFile;
+  onRemove?: () => void;
+}
+
+export default function UploadPdfPreview(props: UploadPdfPreviewProps) {
```

Comment:
i guess this is very similar to the CSV preview? how about extract a common file preview component?

Reply:
ah great. let's add docx and text upload in this PR too